system/ui/raylib/spinner.py

```diff
@@ -0,0 +1,87 @@
+import pyray as rl
+import os
+import select
+import sys
+
+from openpilot.common.basedir import BASEDIR
+from openpilot.system.ui.raylib.gui.application import gui_app
+
+# Constants
+PROGRESS_BAR_WIDTH = 1000
+PROGRESS_BAR_HEIGHT = 20
+ROTATION_RATE = 12.0
+MARGIN = 200
+TEXTURE_SIZE = 360
+FONT_SIZE = 80
+
+def clamp(value, min_value, max_value):
+  return max(min(value, max_value), min_value)
+
+def load_texture_resized(file_name, size):
+  image = rl.load_image(file_name.encode('utf-8'))
+  rl.image_resize(image, size, size)
+  texture = rl.load_texture_from_image(image)
+  rl.unload_image(image)
+  return texture
+
+def check_input_non_blocking():
+  if sys.stdin in select.select([sys.stdin], [], [], 0)[0]:
+    return sys.stdin.readline().strip()
+  return ""
+
+def main():
+  gui_app.init_window("Spinner", fps=30)
```

Comment:
Ideally we want the offroad UI to run at 60FPS

Reply:
to clarify this, we want **all of the UI** to run at 60 fps, but obviously certain things like the camera stream only come in at 20 fps